lib/src/commands.cpp

```diff
@@ -129,146 +129,101 @@ uint8_t message10[] =
+            cap.values[i] = tmp;
         }
+    }
 
-        break;
-      }
-    } break;
-
-    case 4: {
-      switch (type.x) {
-        case 0x5007: {
-          uint16_t aperture;
-          memcpy(&aperture, data + 7, 2);
-          print_hex(data + 4, size - 4);
-          caps.aperture.value = aperture;
-        } break;
-        default:
-          printf("printf_camera_caps_submessage, unknown type (%04X/%04X)\n",
-                 static_cast<int>(type.x), static_cast<int>(type.y));
-        case 0x5012:
-        case 0x500C:
-        case 0x5005:
-        case 0xD001:
-        case 0xD019: {
-          printf("Submessage: ");
-          print_hex(data, size);
-        } break;
-      }
-    } break;
-
-    case 6: {
-      switch (type.x) {
-        case 0xD02A: {
-          const size_t offset =
-              16;  // first is type, next 3 are not ISO levels, unknown
-          if (size >= offset) {
-            caps.iso.numLevels =
-                static_cast<uint32_t>(std::min(size - offset, sizeof(caps.iso.levels)) /
-                sizeof(*caps.iso.levels));
-            memcpy(caps.iso.levels, data + offset,
-                   caps.iso.numLevels * sizeof(*caps.iso.levels));
-          }
-        } break;
-
-        case 0xD240: {
-// don't know what caps.shutter_speed[1] means
-#if 0
-          B: Shutter speed: 1/30s 255 00020200
-          T: Shutter speed: 1/4000s 255 00020280
-          1: Shutter speed: 1/1s 255 00020200
-          2: Shutter speed: 1/2s 255 00020280
-          180x: Shutter speed: 1/180s 255 00020280
-          Shutter speed: 1/250s 255 00020280
-          A: Shutter speed: 1/0s 3 07070000
-#endif
-          printf("Shutter-speed:\n");
-          struct shutter_speed_status {
-            uint32_t unknown0;
-            uint32_t shutter_speed[2];
-            uint32_t unknown1[2];
-          };
-          print_uint32(data + 4, size - 4);
-          print_hex(data + 4, size - 4);
-          assert(size == sizeof(shutter_speed_status) + 4);
-          shutter_speed_status shutterStatus;
-          memcpy(&shutterStatus, data + 4, sizeof(shutterStatus));
-
-          if (static_cast<uint8_t>(shutterStatus.shutter_speed[0]) == 3) {
-            caps.shutter_speed.mode = shutter_speed_auto;
-          } else {
-            caps.shutter_speed.mode = shutter_speed_manual;
-            caps.shutter_speed.value =
-                (shutterStatus.shutter_speed[0] >> 8) / 1000;
-          }
-        } break;
-        case 0xD17C: {
-          printf("Autofocus: ");
-          print_hex(data + 4, size - 4);
-        } break;
-
-        default:
-          printf("printf_camera_caps_submessage, unknown type (%04X/%04X)\n",
-                 static_cast<int>(type.x), static_cast<int>(type.y));
-          break;
-      }
-    } break;
-  }
+    std::string log_capability = std::string("capability: ").append(hex_format(data, size));
+    log(LOG_DEBUG2, log_capability.append(property_strings[cap.property_code]));
```

Comment:
This fails when using a X100F.
An unknown property code (53829) is used with map lookup and string::append does not accept nullptr.
As mentioned elesewhere, we should maybe expose lookup functions instead of exposing the actual map.
Then we could handle unknown properties.